servicenow/getlogs_servicenow-3.8.py

```diff
@@ -0,0 +1,1644 @@
+                    logger.debug('filtered message (exclusion): {} in {}'.format(
+                        filter_val, filter_check))
+                    return
+        logger.debug('passed filter (exclusion)')
+
+    # get project, instance, & device
+    # check_project(get_setting_value(message,
+    #                                'project_field',
+    #                                default=if_config_vars['project_name']),
+    #                                remove=True)
+    logging.debug('message: ' + str(message))
+    instance = get_setting_value(message,
+                                 'instance_field',
+                                 default=UNKNOWN_INSTANCE,
+                                 remove=True)
+
+    if instance != UNKNOWN_INSTANCE and agent_config_vars['instance_regex']:
+        group = re.search(agent_config_vars['instance_regex'], instance)
+        if (group != None):
+            instance = group.group(0)
+    instance = get_alias_from_cache(instance)
+    logger.info(instance)
+    device = get_setting_value(message,
+                               'device_field',
+                               remove=True)
+    # get timestamp
+    try:
+        timestamp = get_setting_value(message,
+                                      'timestamp_field',
+                                      remove=True)
+        timestamp = [timestamp]
+    except ListNotAllowedError as lnae:
+        timestamp = get_setting_value(message,
+                                      'timestamp_field',
+                                      remove=True,
+                                      allow_list=True)
+    except Exception as e:
+        logger.warning(e)
+        sys.exit(1)
+    logger.info(timestamp)
+
+    # get data
+    data = get_data_values(timestamp, message)
+
+    # hand off
+    for timestamp, report_data in list(data.items()):
+        ts = get_timestamp_from_date_string(timestamp)
+        if 'METRIC' in if_config_vars['project_type']:
+            data_folded = fold_up(report_data, value_tree=True)  # put metric data in top level
+            for data_field, data_value in list(data_folded.items()):
+                if data_value is not None:
+                    metric_handoff(
+                        ts,
+                        data_field,
+                        data_value,
+                        instance,
+                        device)
+        else:
+            log_handoff(ts, report_data, instance, device)
+
+
+def label_message(message, fields=[]):
+    """ turns unlabeled, split data into labeled data """
+    if agent_config_vars['data_format'] in {'CSV', 'CSVTAIL', 'IFEXPORT'}:
+        fields = agent_config_vars['csv_field_names']
+    json = dict()
+    for i in range(minlen(fields, message)):
+        json[fields[i]] = message[i]
+    return json
+
+
+def minlen(one, two):
+    return min(len(one), len(two))
+
+
+def parse_csv_message(message):
```

Comment:
It doesn't look like the parse_csv_message method is used at all.  Do we need to keep it? If not, can we remove the parse_csv_data and the parse_csv_row methods too?